docs/uml/macros.iuml

```diff
@@ -42,6 +42,39 @@ skinparam defaultMonospacedFontName Andale Mono
   !endif
 !endif
 
+!ifdef SHOW_DEPLOYMENT_HISTORY
+  !define SHOW_DEPLOYMENT
+
+  resource(DeploymentHistory) {
+    + <u>current</u> (\n\ttype: String,\n\tname: String,\n\trole: Maybe<Object>\n):\n\tMaybe<DeploymentHistory>
+    --
+    + type: String
+    + name: String
+    + role: Maybe<Object>
+    + {method} ref: Deployment
+    + previous: Maybe<DeploymentHistory>
+    ..
+    UNKNOWN_ID
+  }
+
+  DeploymentHistory o-- "1" Deployment
+  DeploymentHistory o-- "0..1" DeploymentHistory
+!endif
+
+!ifdef SHOW_DEPLOYMENT
+  !define SHOW_INSTANCE
+
+  resource(Deployment) {
+    + name: String
+    + role: Maybe<Object>
+    + contractInstances: Array<ContractInstance>
+    ..
+    UNKNOWN_ID
```

Comment:
To go with how we've been doing IDs, I suggest we do a json_stable_stringify of the name and ContractInstance array here and then hash it for the id. Thoughts?

Reply:
In Truffle Teams, we would make this a UUID ideally to conform with the way we identify things. I'm not against supporting a 3rd party ID scheme, but I'm not sure about the flexibility of using the hash of the content as an identifier